### PR TITLE
Scala 3.0.0-M3 -> Scala-3.0.0-RC1

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

**Comment by m-sp** [comment]
Seems circe only supports one version of dotty...

**Comment by rossabaker** [comment]
I think we should probably just give up on RC3.  It's a core dependency, and I don't want to bother with cross-versioning dependencies.

**Comment by rossabaker** [comment]
Core repo dependency, that is, not http4s-core module.  It'll be here after the Great Schism.

**Comment by rossabaker** [comment]
I guess this is the 3.0.0-RC1 branch now.

File: build.sbt [hunk 1]
```diff
@@ -46,8 +46,7 @@ enablePlugins(SonatypeCiReleasePlugin)
 versionIntroduced := Map(
   // There is, and will hopefully not be, an 0.22.0. But this hushes
   // MiMa for now while we bootstrap Dotty support.
-  "3.0.0-M2" -> "0.22.0",
-  "3.0.0-M3" -> "0.22.0",
+  "3.0.0-RC1" -> "0.22.0",
```
Comment (m-sp) [inline build.sbt:49]:
```suggestion
  scala_3 -> "0.22.0",
```

**Review by m-sp — APPROVED** [review]

**Pull request merged**